### PR TITLE
fix: correct log file name formatting error

### PR DESCRIPTION
**Pull request by peternhale**
@ W-13682486@
fix logfile name formatting error introduced by #5277
Ensures the directory to the logfiles is present before saving the log file when log file directories are not present issue [#178](https://github.com/forcedotcom/easywriter/issues/178)

### CODE REVIEW

File: packages/salesforcedx-utils-vscode/test/jest/date/format.test.ts
Comment (peternhale) [inline packages/salesforcedx-utils-vscode/test/jest/date/format.test.ts:1]: New tests

File: packages/salesforcedx-vscode-apex/src/commands/forceAnonApexExecute.ts [hunk 1]
```diff
@@ -168,6 +166,7 @@ export class AnonApexLibraryExecuteExecutor extends LibraryCommandletExecutor<
       return false;
     }
 
+    fs.mkdirSync(path.dirname(logFilePath), { recursive: true });
```
Comment (peternhale) [inline packages/salesforcedx-vscode-apex/src/commands/forceAnonApexExecute.ts:169]: Ensure the `.sfdx/tools/debug/logs` dir is present

Comment (daphne-sfdc) [reply packages/salesforcedx-vscode-apex/src/commands/forceAnonApexExecute.ts:169]: So this line is the fix for https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00001V4nITYAZ/view, right?  This line creates the `.sfdx/tools/debug/logs` directory if it doesn't already exist in the project?

**Review by daphne-sfdc — APPROVED** [review]
Apex Replay Debugger is now working as expected ✅

QE doc here: https://salesforce.quip.com/m3bQA3xym7j4

**Pull request merged**